bot.py

```diff
@@ -1845,6 +1872,87 @@ async def on_guild_channel_update(
                 f"Category {before.id} was updated but is not associated with a role in cache. This could be an error."
             )
 
+@bot.event
+async def on_guild_emojis_update(guild: discord.Guild, before: Sequence[discord.Emoji], after: Sequence[discord.Emoji]):
+    # PLAN:
+    #   - Check if operation is an add/rename/delete
+    #   - Check if operation was done in control server
+    #       - If not, ask control server if synching should be done
+    #           - If yes, sync
+    #           - If no, do nothing
+    #       - If yes, go ahead and synch
+    bot_commands = bot.get_channel(BOT_COMMANDS_ID)
+
+    # Determine if change was made in control server
+    changed_in_hub = guild.id == HUB_SERVER_ID
+
+    # Determine operation and execute
+
+    # Add
+    if len(before) < len(after):
+        emoji = discord.utils.find(lambda e: e not in before, after)
+        if not emoji:
+            Log.error('find() returned None on detected added emoji')
+            return
+        
+        # Automatically sync throughout all guilds if made in control
+        if changed_in_hub:
+            await sync_add(bot=bot, emoji=emoji)
+            
+        # Send View and wait for acceptance or denial
+        else:
+            # TODO: Format based on desired goals
+            await bot_commands.send(
+                f'An emoji, {emoji.name} with the appearence {emoji} has been added. Would you like to sync this change?', 
+                view=EmojiSyncView(emoji=emoji, mod_type='Add')
+            )
+
+    # Delete
+    elif len(before) > len(after):
+        emoji = discord.utils.find(lambda e: e not in after, before)
+        if not emoji:
+            Log.error('find() returned None on detected deleted emoji')
+            return
+        
+        # Auto-sync
+        if changed_in_hub:
+            sync_delete(bot=bot, emoji=emoji)
+
+        # Send View and wait for acceptance or denial
+        else:
+            await bot_commands.send(
+                f'An emoji, {emoji.name} with the appearence {emoji} has been deleted, Would you like to sync this change?',
```

Comment:
Bot still sends this message if the hub server doesn't have the emoji - 

If the hub server doesn't have the emoji (meaning we declined to sync the emoji to all servers when it was uploaded) it should not prompt a message.

Reply:
> Bot still sends this message if the hub server doesn't have the emoji -
> 
> If the hub server doesn't have the emoji (meaning we declined to sync the emoji to all servers when it was uploaded) it should not prompt a message.

Not sure what's going on here. This `else` block says that if the change occurred in a server other than the hub server it should send this message, that is all. Is that not how it's behaving?

Reply:
It's correct - however it shouldn't send the message if the hub server doesn't have the emoji.

So, if something like this happens:
1. Non hub server uploads emoji
2. Hub server declines to sync (meaning the emoji only exists in the non hub server)
3. Non hub server deletes said emoji
It should not send a message to the hub server, because there is nothing to sync. 

It should perform a check before sending this emoji to see if it's a synced emoji. If it's a server specific emoji it should not sync.

Reply:
I see what you're saying. If the emoji doesn't exist in the hub server there is no delete sync to happen. It's disconnected from the initial interaction where the emoji was declined to be uploaded, but basically no change should occur if the emoji isn't already in the hub server. 

Agree with this, check should change to only ask about sync if emoji exists in hub, otherwise ignore deletion (nothing to sync).

Reply:
So just to make sure I understand. If an emoji is deleted and it is not in the hub server, it shouldn't even ask the hub if this is a change that should be made?

Reply:
Correct. If the emoji isn't a synced emoji (as in it's in the hub server and all other servers) it should not create a request to sync if it's deleted. There would be nothing to sync!

Some emojis won't make sense to sync to all servers and will only exist in one non-hub server.

Reply:
Yes. If an emoji is uploaded to a server A that is not hub, and someone selects not to sync it in the hub, then later the emoji is deleted in server A, there is nothing that needs to be done because it was never synced in the first place. Any emojis that can be deleted synchronously will be present in the hub server. Does that make sense?

Reply:
Okay will make that change now. Just have to resolve this error and the creating the emoji. Hopefully should finish it tonight if I have the time.